core/src/main/java/org/mule/runtime/core/api/config/CorrelationIdGenerator.java

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.api.config;
+
+public interface CorrelationIdGenerator {
```

Comment:
why does this need to be api?

Reply:
It is gotten from MuleConfiguration, which is API